mountpoint-s3/src/fs.rs

```diff
@@ -1016,18 +1016,22 @@ where
 
```

Comment:
on line 1010 we have a special case; can you check if removing it does not break the tests?

Reply:
I've added a test for using cached response after a rewind. It works as expected, as the cached response is updated after the first request after rewind completes.

After commenting out the block on [line 1010](https://github.com/awslabs/mountpoint-s3/blob/e2680a231511832472c23c086ecb1493ad73201e/mountpoint-s3/src/fs.rs#L1010C12-L1010C35) only rewind tests are failing, which is expected: 
```
test test_readdir_rewind_with_new_files::default_default_expects ... FAILED
test test_readdir_rewind_ordered ... FAILED
test test_readdir_repeat_response_partial::first_in_the_end_second_in_full ... ok
test test_readdir_then_open_cached ... ok
test test_readdir_repeat_response_partial::first_in_the_beginning_second_in_full ... ok
test test_readdir_repeat_response_partial::first_in_full_second_in_the_end ... ok
test test_readdir_rewind_unordered ... FAILED
test test_readdir_rewind_with_local_files_only ... FAILED
test test_readdir_vs_readdirplus ... ok
test test_readdir_rewind_with_new_files::s3filesystemconfig_s3_personality_s3personality_expressonezone_default_default_expects ... FAILED
test test_readdir_repeat_response_partial::first_in_full_second_in_the_beginning ... ok
```

Reply:
Decided to explicitly destroy the cache on rewind.

(After another look at it I had a suspicion that when rewind happens while cached response starts at offset 0, this response will be repeated. In an unordered case it would have lead to wrong results. It proved to be wrong, as we don't try to use cache in this case as `offset == dir_handle.offset() == 0`, but destroying the cache, while being redundant, protects us from possible bugs in future)